src/test/kotlin/interactor/GetCityHasCheapestInternetConnectionInteractorTest.kt

```diff
@@ -1,32 +1,61 @@
 package interactor
 
-import data.FakeDataSource
-import data.TestCase
+import data.CheapestInternetFakeData
+import data.EmptyFakeData
+import data.InvalidFakeData
 import org.junit.jupiter.api.*
 import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.function.Executable
+import java.lang.IllegalArgumentException
+
 
 
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
 internal class GetCityHasCheapestInternetConnectionInteractorTest {
 
     private lateinit var cityHasCheapestInternet: GetCityHasCheapestInternetConnectionInteractor
-    private lateinit var fakeData: FakeDataSource
+
+    private lateinit var fakeData: CheapestInternetFakeData
+    private lateinit var emptyFakeData: EmptyFakeData
+    private lateinit var invalidFakeData: InvalidFakeData
+
 
     @BeforeAll
     fun setup() {
-        fakeData = FakeDataSource()
+        fakeData = CheapestInternetFakeData()
+        emptyFakeData =EmptyFakeData()
+        invalidFakeData = InvalidFakeData()
+    }
+
+    @Test
+    fun `should return cheapest city when given list of cities`() {
+        // given city entity
         cityHasCheapestInternet = GetCityHasCheapestInternetConnectionInteractor(fakeData)
-        fakeData.changeDataSource(TestCase.CheapestInternet)
+        val expectedCity = fakeData.getAllCitiesData()[9]
+        // when check for the cheapest city in the list
+        val result = cityHasCheapestInternet.execute()
+        // then check the result
+        assertEquals(expectedCity, result)
+    }
+
+    @Test
+    fun `should return null when all city has null internet price`() {
```

Comment:
should return Exception, not null  because your assert function is checked about assertThrows, not null